nixos/modules/services/web-apps/davical.nix

```diff
@@ -0,0 +1,236 @@
+        This is only used once when DAViCal is first initialized.  It is simpler to use this option
+        than the webAdminPassFile, and you can always change the web administrator password once
+        it has been initialized.
+      '';
+    };
+    webAdminPassFile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        The full path to a file on the target system that contains the website administrator password.
+        This file must be readable by the postgres user and will be used to set the web administrator password when
+        DAViCal is first initialized.  If this file cannot be read, the davical-init.service will fail.  If this
+        happens, simply put the file in place and re-run the service to complete initialization.
+      '';
+    };
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        $c->sysabbr     = 'DAViCal';
+        $c->domain_name = "calendar.example.net";
+        $c->admin_email = 'admin@example.net';
+        $c->system_name = "NixOS DAViCal Server";
+      '';
+      description = ''
+        Extra configuration options to add to DAViCal's config.php file. Refer to <link xlink:href="http://wiki.davical.org/w/Configuration_settings"/> for supported
+        values.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = !(cfg.webAdminPass != null && cfg.webAdminPassFile != null);
+        message = "Please specify no more than one of webAdminPass or webAdminPassFile";
+      }
+      {
+        assertion = !(cfg.webAdminPass == "");
+        message = "webAdminPass cannot be empty";
+      }
+    ];
+
+    services.phpfpm.pools.davical = {
+      user = cfg.user;
+      group = config.services.httpd.group;
+      settings = {
+        "listen.owner" = config.services.httpd.user;
+        "listen.group" = config.services.httpd.group;
+        "php_value[include_path]" = "${pkgs.davical}/share/davical/inc:${pkgs.davical.awl}/share/awl/inc";
+        "php_value[magic_quotes_gpc]" = "0";
+        "php_value[register_globals]" = "0";
+        "php_value[error_reporting]" = "E_ALL & ~E_NOTICE";
+        "php_value[default_charset]" = "utf-8";
+      } // cfg.poolConfig;
+    };
+
+    services.postgresql = {
+      enable = true;
+      authentication = ''
+        local ${cfg.dbName} ${cfg.dbAppUser} trust
+        local ${cfg.dbName} ${cfg.dbAdminUser} trust
+      '';
+    };
+
+    environment.etc.davical.source = pkgs.linkFarm "etc-davical" [
+      { name = "config.php";
+        path = pkgs.writeText "config.php" (
+          ''
+            <?php
+              $c->pg_connect[] = 'dbname=${cfg.dbName} port=5432 user=${cfg.dbAppUser}';
+          '' + cfg.extraConfig + "?>"
+        );
+      }
+    ];
+
+    # This service is used to initialize the DAViCal database when it's first installed, and to upgrade the database
```

Comment:
Are you sure you want to automatically run database migrations for the sysadmin? It is generally seen as safer to allow/force the sysadmin to do this manually... Your call.